idaes/models/properties/general_helmholtz/helmholtz_functions.py

```diff
@@ -0,0 +1,2102 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""Generic Helmholtz EOS Functions and Parameters
+"""
+__author__ = "John Eslick"
+
+import enum
+import ctypes
+import math
+
+from matplotlib import pyplot as plt
+import numpy as np
+
+import pyomo.environ as pyo
+from pyomo.core.base.units_container import InconsistentUnitsError
+from pyomo.common.fileutils import find_library
+from pyomo.common.config import ConfigValue, In
+import idaes
+from idaes.core.util.exceptions import ConfigurationError
+from idaes.core import declare_process_block_class
+from idaes.core import (
+    StateBlock,
+    StateBlockData,
+    PhysicalParameterBlock,
+    LiquidPhase,
+    VaporPhase,
+    Phase,
+    Component,
+)
+from idaes.models.properties.general_helmholtz.components import (
+    components as supported_components,
+)
+import idaes.logger as idaeslog
+
+_log = idaeslog.getLogger(__name__)
+
+# make sure the required shared libraries load
+try:
+    _flib_ad = find_library("libadolc")
+    if _flib_ad is None:
+        _flib_ad = find_library("libadolc-2")
+    ctypes.cdll.LoadLibrary(_flib_ad)
+except:
+    _flib_ad = None
+
+try:
+    # When compling these, I don't bother changing the extension based on OS,
+    # so the file name is always ends in .so. It's fine.
+    _flib = find_library("general_helmholtz_external.so")
+    ctypes.cdll.LoadLibrary(_flib)
+except:
+    _flib = None
+
+
+def available():
+    """Returns True if the shared library is installed and loads propertly
+    otherwise returns False
+    """
+    return (_flib is not None) and (_flib_ad is not None)
+
+
+class StateVars(enum.Enum):
```

Comment:
I'm not using strings though.

Reply:
Sorry, missed that. However, this would allow you to assign more meaningful strings for each entry instead of an `int`.

Reply:
Sure, but usually the integer isn't directly used.  You use the constant like AmountBasis.MOLE not 1.  If I switched to strings, they'd still just be something you don't see.

Reply:
Generally, but not necessarily. For one, you could do the following:

```
print(StateVars.PH)

>>> "Pressure, Enthalpy"
```

Hardly critical, but does provide some additional information.

Reply:
I'm not sure why I'm being stubborn on this.  This totally doesn't matter, but the standard enum should be more efficient.  As I understand it the use for string enums is to replace things that used to be strings where really an Enum probably should have been used in the first place.  So things can be a little bit better defined without messing up the API, like checking the domain for config options more easily. Instead of any random string it has to be one defined in the StrEnum. Also printing an Enum doesn't print the number.  Like str(AmountBasis.Mass) is "AmountBasis.Mass" not "2."

Then again I could see the argument that this would be better if I didn't make you use the Enum, and just specify "mass" instead of AmountBasis.MASS then users wouldn't need to import the AmountBasis class.  But then again this sort of thing is probably pretty inconsitent across IDAES.

Ok, so regardless of what's done now.  If we could have it whatever way we want, would the argument ``amount_basis=AmountBasis.MASS`` be better or would ``amount_basis="mass"`` be better?  I guess with the string enum it could be either.